src/fmu/dataio/_model/fields.py

```diff
@@ -404,6 +418,69 @@ class SystemInformation(BaseModel):
     See :class:`OperatingSystem`."""
 
 
+class Tracklog(RootModel):
+    """The ``tracklog`` block contains a record of events recorded on these data.
+    This data object describes the list of tracklog events, in addition to functionality
+    for consturcting a tracklog and adding new records to it.
+    """
+
+    root: List[TracklogEvent]
+
+    def __getitem__(self, item: int) -> TracklogEvent:
+        return self.root[item]
+
+    def __iter__(
+        self,
+    ) -> Any:
+        # Using ´Any´ as return type here as mypy is having issues
+        # resolving the correct type
+        return iter(self.root)
+
+    @classmethod
+    def initialize_metadata_tracklog(cls) -> Tracklog:
+        """Initialize the tracklog object with a list of TracklogEvents
+        with event type 'created'"""
```

Comment:
There is only one event when initializing, so this might be more precise 🙂 

```suggestion
        """Initialize the tracklog object with a list containing one 
        TracklogEvent of type 'created'"""
```